src/arrayMethodJoin.js

```diff
@@ -4,8 +4,25 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
+  [].__proto__.join2 = function(separator = ',') {
     // write code here
+    let phrase = '';
+
+    if (this.length === 0) {
+      return '';
+    }
+
+    for (let i = 0; i < this.length - 1; i++) {
+      if (typeof this[i] === 'number'
+      || typeof this[i] === 'string'
+      || typeof this[i] === 'boolean') {
+        phrase += `${this[i]}` + `${separator}`;
```

Comment:
Here it'll beter to use either interpolation:
```suggestion
        phrase += `${this[i]} ${separator}`;
```
or concatenation:
```suggestion
        phrase += this[i] + separator;
```